README.md

```diff
@@ -31,30 +31,123 @@ yarn add broadcast-channel react-cross-client-router
+change the content of the detail view by simply clicking on any of the two links in the master view.
+You can even close the master view and the currently open detail view will be notified, providing you with a
+convenience link to reopen the master view in the new tab.
+
+
+### Application entry point
+
+```jsx
+import React from 'react'
+import ReactDOM from 'react-dom'
+import BroadcastChannel from 'broadcast-channel'
+import { BrowserRouter } from 'react-router-dom'
+import { ClientRouterProvider, ClientRouterContext } from 'react-cross-client-router'
+
+import { App } from './App'
+
+ReactDOM.render(
+  <BrowserRouter>
+    <ClientRouterProvider
+      channel={new BroadcastChannel('react-cross-tab-router')}
+      storage={window.sessionStorage}
+    >
+      <App />
+    </ClientRouterProvider>
+  </BrowserRouter>,
+  document.getElementById('root')
+);
+```
+
+### App
+
+```jsx
+import React, { Component } from 'react';
+import {
+  ClientRouterContext
+} from 'react-cross-client-router'
+
+import { Master } from './Master'
+import { Detail } from './Detail'
+
+class App extends Component {
+  static contextType = ClientRouterContext;
+  render() {
+    const clientRouter = this.context;
+    return (
+      <div>
+        <h1>react-cross-client-router example</h1>
+        <div>
+          <p>Tab name: {clientRouter.tabId}</p>
+          <p>Connected tabs: {clientRouter.tabs.join(", ")}</p>
+        </div>
+        <Route path="/" exact component={Master} />
+        <Route path="/detail/:id" component={Detail} />
+      </div>
+    );
+  }
+}
+```
+
+### Master
 
 ```jsx
-import React from 'react';
-import BroadcastChannel from 'broadcast-channel';
-import { ClientRouterProvider, ClientLink } from 'react-cross-client-router'
-
-export default () => (
-  <ClientRouterProvider
-    channel={new BroadcastChannel('react-cross-tab-router')}
-    storage={window.sessionStorage}
-  >
-    <ClientLink targetTab="parent" to="/foo">
-      Link parent tab to /foo
-    </ClientLink>
-    <ClientLink targetTab="parent" to="/bar">
-      Link parent tab to /foo
-    </ClientLink>
-    <ClientLink targetTab="child" to="/foo">
-      Link child tab to /foo
-    </ClientLink>
-    <ClientLink targetTab="child" to="/bar">
-      Link child tab to /foo
-    </ClientLink>
-  </ClientRouterProvider>
-)
+import React, { Component } from 'react';
+import {
+  withClientRouter,
+  ClientLink
+} from 'react-cross-client-router'
+
+const Master = withClientRouter(({ clientRouter }) => {
+  return (
+    <div>
+      {clientRouter.tabs.includes("detail") ? (
+        <h2>
+          The detail view tab exists, click an detail link to update
+          the content of the detail view...
+        </h2>
+      ) : (
+        <h2>Click one of the links below to open the corresponding detail view in the new tab</h2>
+      )}
+      <ClientLink targetTab="detail" to={`/detail/1`}>
```

Comment:
the string interpolation was just removed so we can just use `to="/detail/1"`

Reply:
I see more and more code where people just settle down on back ticks everywhere and start liking it. I can use regular string though if you like it more (but then single quotes).